### PR TITLE
Fix importing of ES modules from data URIs

### LINKED ISSUE #3566: Bun can't Dynamically Import with Data URI
### What version of Bun is running?

0.6.13

### What platform is your computer?

Linux 4.4.0-18362-Microsoft x86_64 x86_64

### What steps can reproduce the bug?

```js
import("data:text/javascript;base64," + btoa("export function sayHi() { return 'Hi!'; };")).then((script)=>{
    console.log(script.sayHi());
});
```

### What is the expected behavior?

```
Hi!
```

### What do you see instead?

```
error: FileNotFound reading "data:text/javascript;base64,ZXhwb3J0IGZ1bmN0aW9uIHNheUhpKCkgeyByZXR1cm4gJ0hpISc7IH07"
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by yamalight**
### What does this PR do?

This PR makes Bun correctly load module body when importing modules from data URIs.

Fixes #3566

This is my first attempt at zig / contibuting to Bun. 
I'm almost certain I might've changed things in the wrong places / wrong way - but tests pass, so here it is. 
Please do let me know if there are any issues - would be happy to fix / adjust!

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

- [x] I included a test for the new code, or existing tests cover it
- [x] I ran my tests locally and they pass (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: src/bundler.zig [hunk 1]
```diff
@@ -1300,6 +1301,18 @@ pub const Bundler = struct {
                 break :brk logger.Source.initPathString(path.text, "");
             }
 
+            if (strings.startsWith(path.text, "data:")) {
+                const data_url = DataURL.parseWithoutCheck(path.text) catch |err| {
+                    bundler.log.addErrorFmt(null, logger.Loc.Empty, bundler.allocator, "{s} parsing data url \"{s}\"", .{ @errorName(err), path.text }) catch {};
+                    return null;
+                };
+                const body = data_url.decodeData(bun.default_allocator) catch |err| {
```
Comment (Jarred-Sumner) [inline src/bundler.zig]: `bun.default_allocator` will leak memory here, instead lets use `this_parse.allocator`
```suggestion
                const body = data_url.decodeData(this_parse.allocator) catch |err| {
```

**Comment by Jarred-Sumner** [comment]
Thank you for this. I'm honestly surprised this is all it took to make it work.

File: test/js/node/string-module.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { expect, test } from "bun:test";
+
+test("should import and execute ES module from string", async () => {
```
Comment (Jarred-Sumner) [inline test/js/node/string-module.test.js:3]: Can you add tests for when parsing the module fails that it throws a `BuildMessage` and has a source url line?

**Comment by yamalight** [comment]
@Jarred-Sumner done and done. thanks for your guidance!

**Comment by Jarred-Sumner** [comment]
Thank you

**Comment by http-teapot** [comment]
I am experiencing this issue on the latest version 1.2.5 (and the version before that). I can't figure out where this fix landed, was it shipped?

**Pull request merged**